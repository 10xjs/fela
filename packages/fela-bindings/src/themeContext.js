/* @flow */
import ThemeFactory from './ThemeFactory'
import ThemeProviderFactory from './ThemeProviderFactory'

const defaultTheme = {}

export default function themeFactory(
  createElement: Function,
  createContext: Function
) {
  const { Provider, Consumer } = createContext(defaultTheme)

  return {
    ThemeProvider: ThemeProviderFactory(createElement, Consumer, Provider),
    Theme: ThemeFactory(createElement, Consumer),
  }
}
