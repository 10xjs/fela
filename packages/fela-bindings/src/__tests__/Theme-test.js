import React, { createElement, createContext } from 'react'
import TestRenderer from 'react-test-renderer'

import themeContext from '../themeContext'

const { ThemeProvider, Theme } = themeContext(createElement, createContext)

describe('Using the Theme Component', () => {
  it('should use children as its render method', () => {
    const wrapper = TestRenderer.create(
      <ThemeProvider theme={{ color: 'red' }}>
        <Theme>{theme => <div>{JSON.stringify(theme)}</div>}</Theme>
      </ThemeProvider>
    )

    expect(wrapper.toJSON()).toMatchSnapshot()
  })

  it('should use render as its render method', () => {
    const wrapper = TestRenderer.create(
      <ThemeProvider theme={{ color: 'red' }}>
        <Theme render={theme => <div>{JSON.stringify(theme)}</div>} />
      </ThemeProvider>
    )

    expect(wrapper.toJSON()).toMatchSnapshot()
  })

  it('should merge themes', () => {
    const wrapper = TestRenderer.create(
      <ThemeProvider theme={{ color: 'red' }}>
        <ThemeProvider theme={{ bgColor: 'blue' }}>
          <Theme render={theme => <div>{JSON.stringify(theme)}</div>} />
        </ThemeProvider>
      </ThemeProvider>
    )

    expect(wrapper.toJSON()).toMatchSnapshot()
  })

  it('should replace themes', () => {
    const wrapper = TestRenderer.create(
      <ThemeProvider theme={{ color: 'red' }}>
        <ThemeProvider replace theme={{ bgColor: 'blue' }}>
          <Theme render={theme => <div>{JSON.stringify(theme)}</div>} />
        </ThemeProvider>
      </ThemeProvider>
    )

    expect(wrapper.toJSON()).toMatchSnapshot()
  })
})
