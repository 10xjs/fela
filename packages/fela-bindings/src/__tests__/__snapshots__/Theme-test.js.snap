// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Using the Theme Component should merge themes 1`] = `
<div>
  {"color":"red","bgColor":"blue"}
</div>
`;

exports[`Using the Theme Component should replace themes 1`] = `
<div>
  {"bgColor":"blue"}
</div>
`;

exports[`Using the Theme Component should use children as its render method 1`] = `
<div>
  {"color":"red"}
</div>
`;

exports[`Using the Theme Component should use render as its render method 1`] = `
<div>
  {"color":"red"}
</div>
`;
